### PR TITLE
Feature/settings UI

### PR DESCRIPTION
**Pull request by nikgraf**

### CODE REVIEW

File: apps/app/components/workspace/WorkspaceInvitationList.tsx [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+  List,
+  ListHeader,
+  ListItem,
+  ListText,
+  tw,
+  useIsDesktopDevice,
+  View,
+} from "@serenity-tools/ui";
+import { formatDistance, parseJSON } from "date-fns";
+import { isPast } from "date-fns/esm";
```
Comment (nikgraf) [inline apps/app/components/workspace/WorkspaceInvitationList.tsx:12]: @freuleinkorksi why not use `import { isPast } from 'date-fns'`?

File: package.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     "packages/*"
   ],
   "devDependencies": {
+    "@playwright/test": "^1.27.1",
```
Comment (nikgraf) [inline package.json:10]: @freuleinkorksi can you remove it? this shouldn't be here and is not needed to run the test

File: packages/ui/components/iconButton/IconButton.tsx [hunk 3]
```diff
@@ -65,16 +75,32 @@ export const IconButton = forwardRef((props: IconButtonProps, ref) => {
...
+              isPressed && !isLoading && styles.pressed,
               isFocusVisible && styles.focusVisible,
             ]}
             space={2}
           >
-            <Icon
-              name={name}
-              color={isHovered && !transparent ? "gray-800" : iconColor}
-            />
+            {isLoading ? (
```
Comment (nikgraf) [inline packages/ui/components/iconButton/IconButton.tsx:84]: nice addition!

File: packages/ui/components/listIconText/ListIconText.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { StyleSheet } from "react-native";
+import { tw } from "../../tailwind";
+import { useIsDesktopDevice } from "../../hooks/useIsDesktopDevice/useIsDesktopDevice";
+import { HStack, IStackProps } from "native-base";
+import { Icon, IconNames } from "../icon/Icon";
+import { ListText } from "../listText/ListText";
+
+// TODO either iconName or avatar
```
Comment (nikgraf) [inline packages/ui/components/listIconText/ListIconText.tsx:9]: leftover or will you tackle it later?

File: playwright.config.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
+import type { PlaywrightTestConfig } from "@playwright/test";
```
Comment (nikgraf) [inline playwright.config.ts:1]: can you remove it. we have a config in the `backend` folder where we actually run the playwright config

File: tests-examples/demo-todo-app.spec.ts [hunk 6]
```diff
@@ -0,0 +1,427 @@
+import { test, expect, type Page } from "@playwright/test";
```
Comment (nikgraf) [inline tests-examples/demo-todo-app.spec.ts:1]: please remove this file

File: tests/example.spec.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { test, expect } from "@playwright/test";
```
Comment (nikgraf) [inline tests/example.spec.ts:1]: please remove the file

File: .gitignore [hunk 8]
```diff
@@ -28,3 +28,6 @@ apps/app/webviews/opaque/index.html
 /test-results/
 /playwright-report/
 /playwright/.cache/
+/test-results/
```
Comment (nikgraf) [inline .gitignore:31]: this is a duplication and can be removed :)

**Review by backupbrain — COMMENTED** [review]
Beautiful work.

I changed your "nativeID" for "testID" and debugged the  changes in the end-to-end tests.

File: apps/app/components/workspace/WorkspaceInvitationList.tsx (hunk 1 above)
Comment (freuleinkorksi) [reply apps/app/components/workspace/WorkspaceInvitationList.tsx:12]: good catch! no reason, an auto-import fail

File: package.json (hunk 2 above)
Comment (freuleinkorksi) [reply package.json]: will do

File: packages/ui/components/listIconText/ListIconText.tsx (hunk 4 above)
Comment (freuleinkorksi) [reply packages/ui/components/listIconText/ListIconText.tsx:9]: want to tackle it later, but not sure if it is necessary .. thought of doing an xor, what do you think @nikgraf  ?

Comment (nikgraf) [reply packages/ui/components/listIconText/ListIconText.tsx:9]: yeah xor sounds good, in Typescript it should work with https://basarat.gitbook.io/typescript/type-system/discriminated-unions

```tsx
export type ListIconTextProps = IStackProps & {
  type: "iconName"
  main: string;
  iconName: IconNames;
  secondary?: string;
} | {
  type: "avatar"
  main: string;
  avatar: React.ReactNode;
  secondary?: string;
};
```

**Pull request merged**